tests/contrib/funsor/test_valid_models_sequential_plate.py

```diff
@@ -0,0 +1,112 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+import pytest
+import torch
+
+from pyro.ops.indexing import Vindex
+
+# put all funsor-related imports here, so test collection works without funsor
+try:
+    import funsor
+    import pyro.contrib.funsor
+    from pyroapi import distributions as dist
+    from pyroapi import infer, pyro
+    from tests.contrib.funsor.test_valid_models_enum import assert_ok
```

Comment:
is there a reason you've separated these into multiple files?

Reply:
Mostly ease of review and an attempt to minimize potential merge conflicts